ambry-server/src/main/java/com.github.ambry.server/AmbryRequests.java

```diff
@@ -606,6 +607,33 @@ private void handleAdminRequest(Request request) throws InterruptedException, IO
             }
           }
           break;
+        case ReplicationControl:
+          error = ServerErrorCode.No_Error;
+          metrics.replicationControlRequestQueueTimeInMs.update(requestQueueTime);
+          metrics.replicationControlRequestRate.mark();
+          processingTimeHistogram = metrics.replicationControlResponseQueueTimeInMs;
+          responseQueueTimeHistogram = metrics.replicationControlResponseQueueTimeInMs;
+          responseSendTimeHistogram = metrics.replicationControlResponseSendTimeInMs;
+          requestTotalTimeHistogram = metrics.replicationControlRequestTotalTimeInMs;
+          ReplicationControlAdminRequest replControlRequest =
+              ReplicationControlAdminRequest.readFrom(requestStream, adminRequest);
+          List<? extends PartitionId> partitionIds;
+          if (replControlRequest.getPartitionId() != null) {
+            error = validateRequest(replControlRequest.getPartitionId(), RequestOrResponseType.AdminRequest);
+            partitionIds = Collections.singletonList(replControlRequest.getPartitionId());
+          } else {
+            partitionIds = clusterMap.getAllPartitionIds();
+          }
+          if (!error.equals(ServerErrorCode.Partition_Unknown) && replicationManager.controlReplicationForPartitions(
+              partitionIds, replControlRequest.getOrigins(), replControlRequest.shouldEnable())) {
+            error = ServerErrorCode.No_Error;
+          } else {
+            logger.error("Could not set enable status for replication of {} from {} to {}. Check partition validity and"
+                    + " origins list", partitionIds, replControlRequest.getOrigins(),
+                replControlRequest.shouldEnable());
+            error = error != ServerErrorCode.Partition_Unknown ? ServerErrorCode.Unknown_Error : error;
```

Comment:
error will not be equal to ServerErrorCode.Partition_Unknown at this point. And hence my suggestion.